src/components/schedule-visualizer/doors-overlay.tsx

```diff
@@ -1,62 +1,46 @@
 /**
  * TODO: Need door location, (v1_x, v1_y) only defines the location of the hinge, we also need the
- * length and orientationo of the door to draw it on the map.
+ * length and orientation of the door to draw it on the map.
  */
 
-import { makeStyles } from '@material-ui/core';
 import * as RomiCore from '@osrf/romi-js-core-interfaces';
-import * as L from 'leaflet';
-import React from 'react';
+import React, { createContext, useContext } from 'react';
+import { viewBoxFromLeafletBounds } from '../../util/css-utils';
 import SVGOverlay, { SVGOverlayProps } from './svg-overlay';
 
-const useStyles = makeStyles(() => ({
-  doorMarker: {
-    cursor: 'pointer',
-    pointerEvents: 'auto',
-  },
-}));
-
-interface DoorsOverlayProps extends SVGOverlayProps {
+import Door from './door/door';
+export interface DoorsOverlayProps extends SVGOverlayProps {
   doors: readonly RomiCore.Door[];
+  onDoorClick?(door: RomiCore.Door): void;
 }
 
 export default function DoorsOverlay(props: DoorsOverlayProps): React.ReactElement {
-  const classes = useStyles();
-  const bounds =
-    props.bounds instanceof L.LatLngBounds ? props.bounds : new L.LatLngBounds(props.bounds);
+  const { doors, onDoorClick, ...otherProps } = props;
+  const viewBox = viewBoxFromLeafletBounds(props.bounds);
+  const doorsState = useContext(DoorStateContext);
```

Comment:
I usually avoid using contexts as I feel it creates a dependency and makes the component less isolated. That being said, I'm still quite new to React so it would help if you could explain the advantage of using context here over passing them as props.

Reply:
Using context it's a way to avoid doing something called prop-drilling. 

_Prop Drilling means to pass props multiple levels of components deep in order to reach a particular component that needs those props. And since a parent can’t pass a prop directly to one of its grandchildren it needs to go through other components on its way down.
The problem is that all those other components won’t actually use the prop, they will just pass it down to lower level. This leads to the unnecessary exposure of information and tighter coupling in the code base. Changing the data’s structure or the name of the prop would require modifications in multiple places.
It’s a common occurrence to rename a prop in a component due to name collisions or for the sake of clarity. Doing that a couple of times in the prop drilling chain will only leave us confused and frustrated._

src: https://hackernoon.com/lessons-learned-common-react-code-smells-and-how-to-avoid-them-f253eb9696a4 or https://medium.com/@jtonti/context-vs-prop-drilling-in-react-aa88fff7cd44

E.g:
_With propDrilling: App -> Schedule Visualizer -> DoorOverlay -> Door_
_With context: App -> Door_

Let me know what you think about it.

Hope this helps.

Reply:
Thanks for the great explanation! I have pretty much the same gripes about contexts regarding how it affects the maintainability of the code.

In the example above, the downside I see from context is that even though CH1 and CH2 does not use the context, you cannot render them without a provider (CP) somewhere in the ancestors. This can make the code less readable as there is no where in the definition of CH1 and CH2 that says that they require a context provider, it creates a somewhat hidden dependency that might not be obvious at first glance.

Having a guideline for the use of context sounds like a good idea, but I don't think this project is large enough to need it yet and I don't think I'm experienced enough to write it. But please do share if you know of a good react guideline that we can use.

Reply:
I'm glad that it helps. I'll look for one. :)